src/lib/dependencies/index.js

```diff
@@ -1,73 +1,128 @@
-import { flatten } from 'lodash';
+import cache from '../cache';
+import debug from 'debug';
+import { fetchFileFromRepo, searchFilesFromRepo } from '../github';
+import minimatch from 'minimatch';
 
 import * as composer from './composer';
 import * as npm from './npm';
 import * as nuget from './nuget';
 import * as dep from './dep';
 
+const _debug = debug('dependencies');
+
+const dependencyManagers = {
+  npm,
+  composer,
+  nuget,
+  dep,
+};
+
+const languageToFileType = {
+  JavaScript: 'npm',
+  TypeScript: 'npm',
+  PHP: 'composer',
+  'C#': 'nuget',
+  'Go': 'dep',
+};
```

Comment:
Just to be aware of it, at some point, It's gonna be more complicated than that:

 1) there can be multiple managers for one language: `npm` and `bower` for example in JavaScript
 2) there can be multiple languages for a repo (in GitHub REST API it's just a single value, in GitHub GraphQL API you can already retrieve more)

As you can see in the current `getDependenciesFromGithubRepo`, it was ready to easily handle these cases.

Reply:
Indeed, the previous code was doing a `Promise.all([never_more_than_a_single_promise])` which I didn't reproduce. Are there plans to support `bower` soon? If so I can code it up to support arrays of package managers, should be no problem.

Reply:
I predict the case with multiple languages for a repo (2) will happen sooner than multiple managers for a language (1) . This what I anticipated in `Promise.all(strategies)`.

Both are not absolutely needed now, but we need to have the cases in mind. They should be easy to implement in the new architecture, make sure there is no incompatibility.

A bit of context, at the beginning we were executing all strategies whatever the language, we only started filtering by language to spare resources: overall performance and GitHub API limit.